### PR TITLE
Remove LinearTemperatureProfile, switch to DecayingTemperatureProfile

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

 - Removes `LinearTemperatureProfile` and replaces its use with `DecayingTemperatureProfile`

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: src/Atmos/Model/ref_state.jl [hunk 1]
```diff
@@ -131,10 +111,8 @@ reaches the minimum specified temperature `T_min`
 $(DocStringExtensions.FIELDS)
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:111]: Change documentation above: Replace "A temperature profile that has uniform potential temperature θ until it
reaches the minimum specified temperature `T_min`"

by

A temperature profile that has uniform dry potential temperature θ.

File: test/Numerics/DGmethods/compressible_Navier_Stokes/ref_state.jl [hunk 2]
```diff
@@ -84,15 +84,15 @@ function run2(mpicomm, ArrayType, dim, topl, N, timeend, FT, dt)
         polynomialorder = N,
     )
 
-    T_min, T_s, Γ = FT(290), FT(320), FT(6.5 * 10^-3)
-    RH = 0.01
+    RH = FT(0.01)
```
Comment (tapios) [inline test/Numerics/DGmethods/compressible_Navier_Stokes/ref_state.jl:87]: set `RH=0` here; otherwise, depending how this ref_state is being used, there may be energetic inconsistencies.

File: test/Numerics/DGmethods/compressible_Navier_Stokes/density_current_model.jl [hunk 3]
```diff
@@ -130,10 +130,7 @@ function run(
     model = AtmosModel{FT}(
         AtmosLESConfigType,
         param_set;
-        ref_state = HydrostaticState(
-            DryAdiabaticProfile(typemin(FT), FT(300)),
-            FT(0),
-        ),
+        ref_state = HydrostaticState(DryAdiabaticProfile(FT(300)), FT(0)),
```
Comment (tapios) [inline test/Numerics/DGmethods/compressible_Navier_Stokes/density_current_model.jl:133]: Looks like `DryAdiabaticProfile` only has one argument now, so `FT(0)` I'd think can be omitted as a second argument.

File: test/Diagnostics/diagnostic_fields_test.jl [hunk 4]
```diff
@@ -89,8 +89,7 @@ function config_risingbubble(FT, N, resolution, xmax, ymax, zmax)
 
     # Set up the model
     C_smag = FT(0.23)
-    ref_state =
-        HydrostaticState(DryAdiabaticProfile(typemin(FT), FT(300)), FT(0))
+    ref_state = HydrostaticState(DryAdiabaticProfile(FT(300)), FT(0))
```
Comment (tapios) [inline test/Diagnostics/diagnostic_fields_test.jl:92]: Looks like DryAdiabaticProfile only has one argument now, so FT(0) I'd think can be omitted as a second argument.

File: src/Atmos/Model/AtmosModel.jl [hunk 5]
```diff
@@ -173,11 +169,7 @@ function AtmosModel{FT}(
     param_set::AbstractParameterSet;
     orientation::O = SphericalOrientation(),
     ref_state::RS = HydrostaticState(
-        LinearTemperatureProfile(
-            FT(200),
-            FT(280),
-            FT(grav(param_set)) / FT(cp_d(param_set)),
-        ),
+        DecayingTemperatureProfile(FT(200), FT(60), FT(8 * 10^3)),
```
Comment (tapios) [inline src/Atmos/Model/AtmosModel.jl:172]: Don't you want 
```suggestion
        DecayingTemperatureProfile(FT(290), FT(60), FT(8 * 10^3)),
```
here, i.e., first argument is surface temperature?

File: src/Atmos/Model/AtmosModel.jl [hunk 6]
```diff
@@ -129,11 +129,7 @@ function AtmosModel{FT}(
     param_set::AbstractParameterSet;
     orientation::O = FlatOrientation(),
     ref_state::RS = HydrostaticState(
-        LinearTemperatureProfile(
-            FT(200),
-            FT(280),
-            FT(grav(param_set)) / FT(cp_d(param_set)),
-        ),
+        DecayingTemperatureProfile(FT(200), FT(60), FT(8 * 10^3)),
```
Comment (tapios) [inline src/Atmos/Model/AtmosModel.jl:132]: Same as below: Don't you want 
```suggestion
        DecayingTemperatureProfile(FT(290), FT(60), FT(8 * 10^3)),
```
here, i.e., first argument is surface temperature? (The previous parameters had 280 K, but 290 K is closer to Earth's global-mean temperature.

Comment (tapios) [reply src/Atmos/Model/AtmosModel.jl:132]: The surface temperature, temperature drop, and height scale should probably become CLIMAParameters. The temperatures can be hard-wired parameters. For the height scale, I'd then choose `H = R_d * T_sfc / grav` (which is about 8 km). It would make sense to read the two temperatures from CLIMAParameters, and define the density scale height `H` as a derived parameter.

File: experiments/AtmosLES/risingbubble.jl [hunk 7]
```diff
@@ -214,8 +214,7 @@ function config_risingbubble(FT, N, resolution, xmax, ymax, zmax)
     # The model coefficient for the turbulence closure is defined via the [CLIMAParameters
     # package](https://climate-machine.github.io/CLIMAParameters.jl/latest/)
     # A reference state for the linearisation step is also defined.
-    ref_state =
-        HydrostaticState(DryAdiabaticProfile(typemin(FT), FT(300)), FT(0))
+    ref_state = HydrostaticState(DryAdiabaticProfile(FT(300)), FT(0))
```
Comment (tapios) [inline experiments/AtmosLES/risingbubble.jl:217]: Can't we omit the second argument `FT(0)`?

File: docs/src/ExtendingCLIMA/Atmos/AtmosModel.md [hunk 8]
```diff
@@ -155,10 +155,10 @@ model subcomponent pages to view the possible options for each subcomponent.
     param_set::AbstractParameterSet;
     orientation::O = FlatOrientation(),
     ref_state::RS = HydrostaticState(
-        LinearTemperatureProfile(
+        DecayingTemperatureProfile(
             FT(200),
```
Comment (tapios) [inline docs/src/ExtendingCLIMA/Atmos/AtmosModel.md:159]:
```suggestion
            FT(290),
```
(or 280 K, but 290 K is closer to Earth's global mean). As commented on below, it would be best to make these CLIMAParameters (Earth parameters to be specific).

File: test/Numerics/DGmethods/compressible_Navier_Stokes/density_current_model.jl (hunk 3 above)
Comment (charleskawczynski) [reply test/Numerics/DGmethods/compressible_Navier_Stokes/density_current_model.jl:133]: `DryAdiabaticProfile` does have one argument, what looks like the second argument is actually the second argument for `HydrostaticState`.

File: test/Diagnostics/diagnostic_fields_test.jl (hunk 4 above)
Comment (charleskawczynski) [reply test/Diagnostics/diagnostic_fields_test.jl:92]: `DryAdiabaticProfile` does have one argument, what looks like the second argument is actually the second argument for `HydrostaticState`.

Comment (charleskawczynski) [reply test/Diagnostics/diagnostic_fields_test.jl:92]: I'll move these to separate lines for clarity

File: src/Atmos/Model/AtmosModel.jl (hunk 6 above)
Comment (charleskawczynski) [reply src/Atmos/Model/AtmosModel.jl:132]: Ok, great. I've added an outer convenience constructor that will do this.

File: experiments/AtmosLES/risingbubble.jl (hunk 7 above)
Comment (charleskawczynski) [reply experiments/AtmosLES/risingbubble.jl:217]: Yep, I'll make an outer constructor that defaults to this.

File: src/Atmos/Model/ref_state.jl [hunk 9]
```diff
@@ -42,9 +54,15 @@ struct NoReferenceState <: ReferenceState end
 
 A hydrostatic state specified by a temperature profile and relative humidity.
 """
-struct HydrostaticState{P, F} <: ReferenceState
+struct HydrostaticState{P, FT} <: ReferenceState
     temperatureprofile::P
-    relativehumidity::F
+    relativehumidity::FT
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:59]:
```suggestion
    relative_humidity::FT
```
(change throughout)

**Comment by charleskawczynski** [comment]
bors r-

Going to let #1055 pass through

**Comment by charleskawczynski** [comment]
@bischtob you might want to review some of the experiments after this lands

File: tutorials/Atmos/heldsuarez.jl [hunk 10]
```diff
@@ -138,25 +138,27 @@ FT = Float32
...
+# with each other.
 # In this example, the reference temperature field smoothly transitions from a
 # linearly decaying profile near the surface to a constant temperature profile at the top
 # of the domain.
 T_surface = FT(290) ## surface temperature (K)
 ΔT = FT(60)  ## temperature drop between surface and top of atmosphere (K)
-H_t = FT(8e3) ## height scale over which temperature drops (m)
+_R_d = FT(R_d(param_set))
+_grav = FT(grav(param_set))
+H_t = _R_d * T_surface / _grav
```
Comment (tapios) [inline tutorials/Atmos/heldsuarez.jl:159]: Remove redundant `H_t` definition here too, if that's the default in ref_state.

File: src/Atmos/Model/ref_state.jl [hunk 11]
```diff
@@ -210,31 +153,41 @@ Tv(z) = \\max(Tv{\\text{surface}} − ΔTv \\tanh(z/H_{\\text{t}})
...
     H_t::FT
 end
+function DecayingTemperatureProfile(
+    param_set::AbstractParameterSet,
+    T_virt_surf::FT,
+    ΔTv::FT,
+) where {FT <: AbstractFloat}
+    _R_d::FT = R_d(param_set)
+    _grav::FT = grav(param_set)
+    H_t = _R_d * T_virt_surf / _grav
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl:173]: > 1. Have H_t as a genuine input parameter of DecayingTemperatureProfile

I thought that you asked for `H_t` to be derived from the other arguments?

> 2. Make surface T_surf_ref and ΔTv_ref CIMAParameters for Earth.

Opened [18](https://github.com/CliMA/CLIMAParameters.jl/pull/18) in CLIMAParameters

> Then set DecayingTemperatureProfile with `T_virt_surf = T_surf_ref, ΔTv = ΔTv_ref, and H_t = _R_d * T_surf_ref / grav` on the driver level.

This seems to conflict with a different comment you left:

> If the default behavior for DecayingTemperatureProfile is to use `H_t = _R_d * T_sfc / _grav`, when no `H_t` is given, we can simply omit it here. No need for redundant specifications.

Maybe we can review this interactively?

File: test/Driver/gcm_driver_test.jl [hunk 12]
```diff
@@ -65,10 +65,14 @@ function main()
     timeend = FT(1800)
     # Timestep size (s)
     dt = FT(600)
+    _R_d = FT(R_d(param_set))
+    _grav = FT(grav(param_set))
+    H_t = _R_d * setup.T_ref / _grav
 
     setup = AcousticWaveSetup{FT}()
+    T_profile = IsothermalProfile(setup.T_ref, H_t)
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply test/Driver/gcm_driver_test.jl:73]: If I remember correctly, we said that `IsothermalProfile` would simply call `DecayingTemperatureProfile` with `ΔTv=0`, which is what this is doing. `DecayingTemperatureProfile` requires
 - `T_virt_surf`
 - `ΔTv`
 - `H_t`

We could change this to

```julia
IsothermalProfile(param_set, T::FT) where {FT} = DecayingTemperatureProfile(param_set, T, FT(0))
```
Do we want that interface instead? (Then `H_t` will be computed from `H_t = _R_d * T / _grav`) My original impression was that you wanted this parameter exposed at the driver level.

[the remark this answers is not in the document]
Comment (tapios) [reply test/Driver/gcm_driver_test.jl:73]: It just seems silly to have to define a height scale `H_t` on the driver level, which is never really needed. However you remove it is fine with me, but it needs to go. It creates unnecessary confusion for users.

I do want the option of setting `H_t` to something other than the density scale height when needed. But it's not needed for isothermal.

File: src/Atmos/Model/ref_state.jl (hunk 11 above)
[the remark this answers is not in the document]
Comment (tapios) [reply src/Atmos/Model/ref_state.jl:173]: > > 1. Have H_t as a genuine input parameter of DecayingTemperatureProfile
> 
> I thought that you asked for `H_t` to be derived from the other arguments?

The easiest option, and I thought the one we agreed upon, is to have `H_t = R_d * T_surf/g` when nothing else is given (default), but have the option to override that when needed.

> > 1. Make surface T_surf_ref and ΔTv_ref CIMAParameters for Earth.
> 
> Opened [18](https://github.com/CliMA/CLIMAParameters.jl/pull/18) in CLIMAParameters
> 
> > Then set DecayingTemperatureProfile with `T_virt_surf = T_surf_ref, ΔTv = ΔTv_ref, and H_t = _R_d * T_surf_ref / grav` on the driver level.

Yes, my original suggestion was to set all three as parameters. Then you implemented the scale height in the `DecayingTemperatureProfile`, which is a fine alternative if it is an *option* (not a necessity) there. So then I said just specify `T_virt_surf` and `ΔTv` as CLIMAParameters. 
> > If the default behavior for DecayingTemperatureProfile is to use `H_t = _R_d * T_sfc / _grav`, when no `H_t` is given, we can simply omit it here. No need for redundant specifications.

The statement here is that if `DecayingTemperatureProfile` uses  `H_t = _R_d * T_sfc / _grav` as the default when no `H_t` is given, there is no point to specify in multiple places
`H_t = _R_d * T_sfc / _grav` as a (redundant) input argument.

So in short, here's what works: write the code with `H_t` as optional input argument. Without it, it defaults to the scale height RT/g. But this can be overridden by other height scales specified by a user. 

Interactive would be good. But I'm really busy tonight and all of tomorrow. If the concepts are unclear, perhaps @szy21 can help explain.

File: src/Atmos/Model/ref_state.jl [hunk 13]
```diff
@@ -17,6 +16,19 @@ condition or for linearization.
 """
 abstract type ReferenceState end
 
+"""
+    TemperatureProfile
+
+Specifies the temperature profile for a reference state.
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:22]:
```suggestion
Specifies the temperature virtual temperature profile for a reference state.
```

File: src/Atmos/Model/ref_state.jl [hunk 14]
```diff
@@ -17,6 +16,19 @@ condition or for linearization.
...
+
+Instances of this type are required to be callable objects with the following signature
+
+    T,p = (::TemperatureProfile)(orientation::Orientation, aux::Vars)
+
+where `T` is the temperature (in K), and `p` is the pressure (in hPa).
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:28]:
```suggestion
where `T` is the temperature or virtual temperature (in K), and `p` is the pressure (in hPa).
```

File: src/Atmos/Model/ref_state.jl (hunk 11 above)
[the remark this answers is not in the document]
Comment (szy21) [reply src/Atmos/Model/ref_state.jl:173]: I'm happy to talk tonight if needed.

File: src/Atmos/Model/ref_state.jl [hunk 15]
```diff
@@ -57,15 +77,15 @@ function atmos_init_aux!(
...
+    T, p = m.temperature_profile(atmos.orientation, atmos.param_set, aux)
     FT = eltype(aux)
     _R_d::FT = R_d(atmos.param_set)
 
     aux.ref_state.T = T
     aux.ref_state.p = p
     aux.ref_state.ρ = ρ = p / (_R_d * T)
     q_vap_sat = q_vap_saturation(atmos.param_set, T, ρ)
-    aux.ref_state.ρq_tot = ρq_tot = ρ * m.relativehumidity * q_vap_sat
+    aux.ref_state.ρq_tot = ρq_tot = ρ * m.relative_humidity * q_vap_sat
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:88]:
```suggestion
    #FIXME q_vap_sat needs to be computed with the actual temperature for complete consistency; if T is virtual temperature, this is only approximately the correct q_tot.
    aux.ref_state.ρq_tot = ρq_tot = ρ * m.relative_humidity * q_vap_sat
```

File: src/Atmos/Model/ref_state.jl (hunk 14 above)
Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl]: I did a search-all, only saw `hPa` here and in EDMF documentation, which I just removed since it was a bit more of a reference at the time

**Comment by charleskawczynski** [comment]
@tapios I'm getting a domain error in `saturation_vapor_pressure` in the `experiments/AtmosLES/risingbubble.jl` experiment:
```
ERROR: LoadError: TaskFailedException:
DomainError with -0.0768063663403528:
Exponentiation yielding a complex result requires a complex argument.
Replace x^y with (x+0im)^y, Complex(x)^y, or similar.
Stacktrace:
 [1] throw_exp_domainerror at ./math.jl:36 [inlined]
 [2] overdub(::Cassette.Context{nametype(CPUCtx),KernelAbstractions.CompilerMetadata{KernelAbstractions.NDIteration.DynamicSize,KernelAbstractions.NDIteration.NoDynamicCheck,CartesianIndex{1},CartesianIndices{1,Tuple{Base.OneTo{Int64}}},KernelAbstractions.NDIteration.NDRange{1,KernelAbstractions.NDIteration.DynamicSize,KernelAbstractions.NDIteration.StaticSize{(125,)},CartesianIndices{1,Tuple{Base.OneTo{Int64}}},Nothing}},Nothing,KernelAbstractions.var"##PassType#422",Nothing,Cassette.DisableHooks}, ::typeof(Base.Math.throw_exp_domainerror), ::Float64) at /Users/charliekawczynski/.julia/packages/Cassette/158rp/src/overdub.jl:0
 [3] ^ at ./math.jl:849 [inlined]
 [4] saturation_vapor_pressure at /Users/charliekawczynski/Dropbox/Caltech/work/dev/CliMA/CLIMA/src/Common/MoistThermodynamics/relations.jl:661
```

with these changes. I'm not really sure how to move forward here..

**Comment by charleskawczynski** [comment]
The other issue we're running into is in `experiments/AtmosLES/bomex.jl`:
```
Test Failed at Test FailedTest Failed at  at /central/scratchio/esm/slurmci/sources/e9b5da482cea921e874e18197c5f03c3432d9e33/experiments/AtmosLES/bomex.jl:525
/central/scratchio/esm/slurmci/sources/e9b5da482cea921e874e18197c5f03c3432d9e33/experiments/AtmosLES/bomex.jl:525
/central/scratchio/esm/slurmci/sources/e9b5da482cea921e874e18197c5f03c3432d9e33/experiments/AtmosLES/bomex.jl:525
  Expression:   Expression:   Expression: isapprox(result, FT(1); atol=isapprox(result, FT(1); atol=isapprox(result, FT(1); atol=0.002)
   Evaluated: isapprox(1.0032519f0, 1.0f0; atol=0.002)
0.0.002)
   Evaluated: isapprox(1.0032519f0, 1.0f0; atol=0.002)
002)
   Evaluated: isapprox(1.0032519f0, 1.0f0; atol=0.002)
```
I'm not exactly sure what this test is doing, @akshaysridhar can/should we bump this to `atol=0.004`?

**Comment by charleskawczynski** [comment]
Extending [this remark](https://github.com/CliMA/ClimateMachine.jl/pull/1057#issuecomment-626020873), this is occurring due to Newton method exploring negative temperatures. This is a hard block. Perhaps it would be useful to add a `Bracket(0, Inf)` argument to `RootSolvers.jl` so that we can capture the input values causing this poor search?

**Comment by tapios** [comment]
> Extending [this remark](https://github.com/CliMA/ClimateMachine.jl/pull/1057#issuecomment-626020873), this is occurring due to Newton method exploring negative temperatures. This is a hard block. Perhaps it would be useful to add a `Bracket(0, Inf)` argument to `RootSolvers.jl` so that we can capture the input values causing this poor search?



> Extending [this remark](https://github.com/CliMA/ClimateMachine.jl/pull/1057#issuecomment-626020873), this is occurring due to Newton method exploring negative temperatures. This is a hard block. Perhaps it would be useful to add a `Bracket(0, Inf)` argument to `RootSolvers.jl` so that we can capture the input values causing this poor search?

Yes, negative temperatures would be good to avoid. But I don't understand how this is happening here if it wasn't happening before. You have not changed anything about the saturation adjustment itself, right, just the convergence criterion. So why does it fail now if it wasn't failing before?

**Comment by charleskawczynski** [comment]
>  You have not changed anything about the saturation adjustment itself, right, just the convergence criterion.

The residual-based tolerance is not changed in this PR, it's being changed in #1059. I think we wanted to make these changes one step at a time.

> So why does it fail now if it wasn't failing before?

I'm not sure. Somehow the solution is moving into a thermodynamic state that we're not testing.

File: docs/src/HowToGuides/Atmos/Model/risingbubble.md [hunk 16]
```diff
@@ -297,8 +297,8 @@ package](https://CliMA.github.io/CLIMAParameters.jl/latest/)
 A reference state for the linearisation step is also defined.
 
 ```@example risingbubble
-    ref_state =
-        HydrostaticState(DryAdiabaticProfile(typemin(FT), FT(300)), FT(0))
+    T_profile = DryAdiabaticProfile(FT(290))
```
Comment (tapios) [inline docs/src/HowToGuides/Atmos/Model/risingbubble.md:300]:
```suggestion
    T_profile = DryAdiabaticProfile(FT(300))
```
Leave this at 300 K for this test case. Not sure if that causes the problems, but it might.

Comment (charleskawczynski) [reply docs/src/HowToGuides/Atmos/Model/risingbubble.md:300]: This is the docs markdown for RB. I can try this change in the risingbuble.jl, though.

File: experiments/AtmosLES/risingbubble.jl [hunk 17]
```diff
@@ -214,8 +214,8 @@ function config_risingbubble(FT, N, resolution, xmax, ymax, zmax)
     # The model coefficient for the turbulence closure is defined via the [CLIMAParameters
     # package](https://CliMA.github.io/CLIMAParameters.jl/latest/)
     # A reference state for the linearisation step is also defined.
-    ref_state =
-        HydrostaticState(DryAdiabaticProfile(typemin(FT), FT(300)), FT(0))
+    T_profile = DryAdiabaticProfile(FT(290))
```
Comment (tapios) [inline experiments/AtmosLES/risingbubble.jl:217]:
```suggestion
    T_profile = DryAdiabaticProfile(FT(300))
```

**Comment by tapios** [comment]
> > You have not changed anything about the saturation adjustment itself, right, just the convergence criterion.
> 
> The residual-based tolerance is not changed in this PR, it's being changed in #1059. I think we wanted to make these changes one step at a time.
> 
> > So why does it fail now if it wasn't failing before?
> 
> I'm not sure. Somehow the solution is moving into a thermodynamic state that we're not testing.

The dry adiabatic profile used to cut off at 0 K in the rising thermal bubble, now it doesn't. It seems it shouldn't get cold in this case, but there may be an issue there. It may be worthwhile adding a temperature cutoff back in for the dry adiabatic case, to avoid too low temperatures in the reference state.

File: src/Atmos/Model/ref_state.jl (hunk 11 above)
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl:177]: Ok, yes, sorry for the confusion. Now, there is a single `DecayingTemperatureProfile` method with optional arguments

 - T_virt_surf
 - ΔTv
 - H_t

Once ΔTv and T_virt_surf are merged into CLIMAParameters, I'll make these values default to those defined in CLIMAParameters.

**Comment by charleskawczynski** [comment]
bors try

File: src/Atmos/Model/ref_state.jl [hunk 18]
```diff
@@ -138,11 +147,10 @@ function (profile::DryAdiabaticProfile)(
     z = altitude(orientation, param_set, aux)
 
     # Temperature
-    T = profile.T_surface - _grav * z / _cp_d
+    T = max(profile.T_surface - _grav * z / _cp_d, profile.T_min)
 
     # Pressure
     p = _MSLP * (1 - _grav * z / (_cp_d * profile.T_surface))^(_cp_d / _R_d)
-
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:182]: You need to make the pressure consistent with the temperature profile (i.e., replace with the pressure profile that you had before that included the temperature cutoff.

File: src/Atmos/Model/ref_state.jl [hunk 19]
```diff
@@ -3,11 +3,10 @@ using DocStringExtensions
 export NoReferenceState,
     HydrostaticState,
     IsothermalProfile,
-    LinearTemperatureProfile,
     DecayingTemperatureProfile,
     DryAdiabaticProfile
 
-using CLIMAParameters.Planet: R_d, MSLP, cp_d, grav
+using CLIMAParameters.Planet: R_d, MSLP, cp_d, grav, T_min
```
Comment (tapios) [inline src/Atmos/Model/ref_state.jl:9]:
```suggestion
using CLIMAParameters.Planet: R_d, MSLP, cp_d, grav, T_min_ref
```
(And similarly throughout. It’s important to be clear that this is a reference state minimum temperature, not a general minimum temperature.)

File: src/Atmos/Model/ref_state.jl (hunk 18 above)
Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl:182]: I thought this might have been needed and was going to ask before merging. Thanks!

File: src/Atmos/Model/ref_state.jl (hunk 19 above)
Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl:9]: Ok, I don't think we've added `T_min_ref` to CLIMAParameters yet. I'll add this

Comment (charleskawczynski) [reply src/Atmos/Model/ref_state.jl:9]: Opened [PR 20](https://github.com/CliMA/CLIMAParameters.jl/pull/20) in CLIMAParameters @tapios

**Comment by charleskawczynski** [comment]
Closes #939, #882

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
@tapios it looks like everything is fixed except a domain error in saturation vapor pressure in `tutorials/Atmos/risingbubble.jl`, which is based on the DryAdiabaticProfile. I visually verified in a new branch: [ck/RemoveProfilesModularize](https://github.com/CliMA/ClimateMachine.jl/tree/ck/RemoveProfilesModularize). But, based on the doc string

"A temperature profile that has uniform dry potential temperature `θ`"

I also included `θ_dry` via

```julia
dry_adiabatic = DryAdiabaticProfile{FT}(param_set);
args = dry_adiabatic.(Ref(param_set), z)
T = first.(args)
p = last.(args)
θ_dry = dry_pottemp_given_pressure.(Ref(param_set), T, p) # assuming that this is temperature, and not virtual temperature
```

dry_adiabatic

I think it might be worth opening a PR on the `ck/RemoveProfilesModularize` branch, since it's easier to include plots like this.

**Comment by charleskawczynski** [comment]
bors try
